kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/DataServiceImpl.java

```diff
@@ -680,4 +650,99 @@ public CloudConnectionStatusEnum getNotificationStatus() {
     public void setNotificationStatus(CloudConnectionStatusEnum status) {
         this.notificationStatus = status;
     }
+
+    private final class PublishManager implements Runnable {
+
+        @Override
+        public void run() {
+            Thread.currentThread().setName("DataServiceImpl:Submit");
+            while (publisherEnabled) {
+                long sleepingTime = -1;
+                if (DataServiceImpl.this.dataTransportService.isConnected()) {
+                    try {
+                        DataMessage message = DataServiceImpl.this.store.getNextMessage();
+
+                        if (message != null) {
+                            checkInFlightMessages(message);
+
+                            boolean messagePublished;
+                            if (dataServiceOptions.isRateLimitEnabled() && message.getPriority() >= 5) {
+                                messagePublished = publishMessageTokenBucket(message);
+                            } else {
+                                messagePublished = publishMessageUnbound(message);
```

Comment:
this method alway returns true so set messagePublished to true explicitly or intialize it to true.